### ISSUE TITLE
emqttd_ctl can't connect unless broker name resolves to localhost

### ISSUE THREAD

**Issue by victorclaessen** [emqx/emqx#2153]
#### Environment

- OS: Debian 9
- Erlang/OTP: 19
- EMQ: 2.10

#### Description

emqttd_ctl cannot connect to the broker node when the dns name doesn't resolve to localhost.

I have a node at emq@myserver.mydomain.nl (hostname simplifed for clarity). I can use emqttd_ctl, as long as I have in my /etc/hosts the following line:
`127.0.1.1       myserver.mydomain.nl myserver`

When I change this to the actual ip (also simplifed for clarity):
`10.0.0.1          myserver.mydomain.nl myserver`
I cannot connect anymore with emqttd_ctl:
**`Node 'emq@myserver.mydomain.nl' not responding to pings.`**

I have turned off my firewall for testing, and have also changed emq.conf to allow connections the api on all interfaces, but that did not make a difference.
`#listener.api.mgmt = 127.0.0.1:8088`
`listener.api.mgmt = 0.0.0.0:8088`

How, exactly, does emqttd_ctl connect to the broker node?

**Comment by linjunjj** [comment]
Hi @victorclaessen  Did you modify the corresponding host name  in 'node. name' of  'emqx. conf'?

**Comment by victorclaessen** [comment]
It is.

**Comment by victorclaessen** [comment]
Perhaps we can first address my question, and then figure out where things go wrong?
> How, exactly, does emqttd_ctl connect to the broker node?

**Comment by linjunjj** [comment]
@victorclaessen  connect to the broker node by RPC  
`relx_nodetool rpc emqx_ctl run_command $@`

**Comment by linjunjj** [comment]
@victorclaessen  When you use the emqx_ctl command,  Will start [EPMD](http://erlang.org/doc/man/epmd.html). Epmd completes mapping between Erlang nodes and IP ports,In fact, Start a node will be connected to the broker  node. But this is a brief connection. Next send commands to emqx broker by RPC，When `myserver.mydomain.nl myserver` points to a local ip, it is actually usable. I feeling that  the `10.0.0.1 `  is not the native actual IP, This IP feels like a router, Maybe you should check  IP.

**Comment by victorclaessen** [comment]
The ip address was simplified, with the intended effect of making things simpler. If you must know, the actual ip is 192.168.14.214, and it is definitely on my nic.

```
root@myserver:/# ip a | grep "\.14.214"
    inet 192.168.14.214/24 brd 192.168.14.255 scope global eth814
```

And it also resolves nicely:
```
root@myserver:/# nslookup myserver.mydomain.nl
Server:         192.168.14.1
Address:        192.168.14.1#53

Name:   myserver.mydomain.nl
Address: 192.168.14.214
```

And in /etc/hosts:
```
root@myserver:/# cat /etc/hosts
127.0.0.1       localhost
#127.0.1.1      myserver.mydomain.nl myserver
127.0.1.1       myserver
192.168.14.224  myserver.mydomain.nl
```

epmd is running, and is listening on all interfaces:
```
root@myserver:/# netstat -lpan | grep epmd | grep LISTEN
tcp        0      0 0.0.0.0:4369            0.0.0.0:*               LISTEN      10696/epmd
tcp6       0      0 :::4369                 :::*                    LISTEN      10696/epmd
```

The firewall is off:
```
root@myserver:/# ufw status
Status: inactive
```

epmd can find the local node:
```
root@myserver:/## epmd -names
epmd: up and running on port 4369 with data:
name emq at port 6369
```

But:
```
root@myserver:/## emqttd_ctl
Node 'emq@myserver.mydomain.nl' not responding to pings.
```

**Comment by Gilbert-Wong** [comment]
Hi, @victorclaessen , execute `ps aux | grep emq` and please show me the result. And please show me the `etc/emq.conf` which you have modified. I would help you resolve this issue.

**Comment by victorclaessen** [comment]
ps aux | grep emq
```
emqtt    20486  0.0  0.1  19872  1688 ?        S    13:44   0:00 /usr/lib/emqttd/erts-9.0/bin/run_erl -daemon //var/lib/emqttd/emqtt_erl_pipes/emq@myserver.mydomain.nl/ /var/log/emqttd exec "/usr/sbin/emqttd" "console"
emqtt    20488  0.2  6.4 5622988 65660 pts/0   Ssl+ 13:44   0:02 /usr/lib/emqttd/erts-9.0/bin/beam.smp -W w -e 256000 -Q 65536 -P 256000 -A 32 -K true -zdbbl 32768 -- -root /usr/lib/emqttd -progname usr/sbin/emqttd -- -home /var/lib/emqttd -- -boot /usr/lib/emqttd/releases/2.3.10/emqttd -mode embedded -boot_var ERTS_LIB_DIR /usr/lib/emqttd/erts-9.0/../lib -mnesia dir "/var/lib/emqttd/mnesia/emq@myserver.mydomain.nl" -config /var/lib/emqttd/configs/app.2019.01.21.13.44.11.config -kernel net_ticktime 60 -smp auto -setcookie mysecretcookie -proto_dist inet_tcp -name emq@myserver.mydomain.nl -vm_args /var/lib/emqttd/configs/vm.2019.01.21.13.44.11.args -- console
emqtt    20817  0.0  0.0   4180   656 ?        Ss   13:44   0:00 erl_child_setup 1024
emqtt    20893  0.0  0.0   4288   764 ?        Ss   13:44   0:00 sh -s disksup
emqtt    20894  0.0  0.0   4044   636 ?        Ss   13:44   0:00 /usr/lib/emqttd/lib/os_mon-2.4.2/priv/bin/memsup
emqtt    20895  0.0  0.0   4176   664 ?        Ss   13:44   0:00 /usr/lib/emqttd/lib/os_mon-2.4.2/priv/bin/cpu_sup
root     23255  0.0  0.0  12784   948 pts/3    S+   14:03   0:00 grep emq
```

cat /etc/emqttd/emq.conf
```
##===================================================================
## EMQ Configuration R2.3
##===================================================================

##--------------------------------------------------------------------
## Cluster
##--------------------------------------------------------------------

## Cluster name
cluster.name = emqcl

## Cluster discovery strategy: manual | static | mcast | dns | etcd | k8s
cluster.discovery = mcast

## Cluster Autoheal: on | off
cluster.autoheal = on

## Clean down node of the cluster
cluster.autoclean = 5m

##--------------------------------------------------------------------
## Cluster with static node list

## cluster.static.seeds = emq1@127.0.0.1,emq2@127.0.0.1

##--------------------------------------------------------------------
## Cluster with multicast

cluster.mcast.addr = 239.192.0.1
cluster.mcast.ports = 4369,4370
cluster.mcast.iface = 0.0.0.0
cluster.mcast.ttl = 255
cluster.mcast.loop = on

##--------------------------------------------------------------------
## Cluster with DNS

## cluster.dns.name = localhost

## cluster.dns.app = emq

##--------------------------------------------------------------------
## Cluster with Etcd

## cluster.etcd.server = http://127.0.0.1:2379

## cluster.etcd.prefix = emqcl

## cluster.etcd.node_ttl = 1m

##--------------------------------------------------------------------
## Cluster with k8s

## cluster.k8s.apiserver = http://10.110.111.204:8080

## cluster.k8s.service_name = emq

## Address Type: ip | dns
## cluster.k8s.address_type = ip

## The Erlang application name 
## cluster.k8s.app_name = emq

##--------------------------------------------------------------------
## Node Args
##--------------------------------------------------------------------

## Node name
node.name = emq@myserver.mydomain.nl

## Cookie for distributed node
node.cookie = mysecretcookie

## SMP support: enable, auto, disable
node.smp = auto

## vm.args: -heart
## Heartbeat monitoring of an Erlang runtime system
## Value should be 'on' or comment the line
## node.heartbeat = on

## Enable kernel poll
node.kernel_poll = on

## async thread pool
node.async_threads = 32

## Erlang Process Limit
node.process_limit = 256000

## Sets the maximum number of simultaneously existing ports for this system
node.max_ports = 65536

## Set the distribution buffer busy limit (dist_buf_busy_limit)
node.dist_buffer_size = 32MB

## Max ETS Tables.
## Note that mnesia and SSL will create temporary ets tables.
node.max_ets_tables = 256000

## Tweak GC to run more often
node.fullsweep_after = 1000

## Crash dump
node.crash_dump = /var/log/emqttd/crash.dump

## Distributed node ticktime
node.dist_net_ticktime = 60

## Distributed node port range
node.dist_listen_min = 6369
node.dist_listen_max = 6379

##--------------------------------------------------------------------
## Log
##--------------------------------------------------------------------

## Set the log dir
log.dir = /var/log/emqttd

## Console log. Enum: off, file, console, both
log.console = console

## Console log level. Enum: debug, info, notice, warning, error, critical, alert, emergency
log.console.level = error

## Syslog. Enum: on, off
log.syslog = on

##  syslog level. Enum: debug, info, notice, warning, error, critical, alert, emergency
log.syslog.level = error

## Console log file
## log.console.file = /var/log/emqttd/console.log

## Info log file
## log.info.file = /var/log/emqttd/info.log

## Error log file
log.error.file = /var/log/emqttd/error.log

## Enable the crash log. Enum: on, off
log.crash = on

log.crash.file = /var/log/emqttd/crash.log

##--------------------------------------------------------------------
## Allow Anonymous and Default ACL
##--------------------------------------------------------------------

## Allow Anonymous authentication
mqtt.allow_anonymous = true

## ACL nomatch
mqtt.acl_nomatch = allow

## Default ACL File
mqtt.acl_file = /etc/emqttd/acl.conf

## Cache ACL for PUBLISH
mqtt.cache_acl = true

##--------------------------------------------------------------------
## MQTT Protocol
##--------------------------------------------------------------------

## Max ClientId Length Allowed.
mqtt.max_clientid_len = 1024

## Max Packet Size Allowed, 64K by default.
mqtt.max_packet_size = 64KB

## Check Websocket Protocol Header. Enum: on, off
mqtt.websocket_protocol_header = on

## The Keepalive timeout: Keepalive * backoff * 2
mqtt.keepalive_backoff = 1.25

##--------------------------------------------------------------------
## MQTT Connection
##--------------------------------------------------------------------

## Force GC: integer. Value 0 disabled the Force GC.
mqtt.conn.force_gc_count = 100

##--------------------------------------------------------------------
## MQTT Client
##--------------------------------------------------------------------

## Client Idle Timeout (Second)
mqtt.client.idle_timeout = 30s

## Max publish rate of Messages
## mqtt.client.max_publish_rate = 5

## Enable client Stats: on | off
mqtt.client.enable_stats = off

##--------------------------------------------------------------------
## MQTT Session
##--------------------------------------------------------------------

## Max Number of Subscriptions, 0 means no limit.
mqtt.session.max_subscriptions = 0

## Upgrade QoS?
mqtt.session.upgrade_qos = off

## Max Size of the Inflight Window for QoS1 and QoS2 messages
## 0 means no limit
mqtt.session.max_inflight = 32

## Retry Interval for redelivering QoS1/2 messages.
mqtt.session.retry_interval = 20s

## Client -> Broker: Max Packets Awaiting PUBREL, 0 means no limit
mqtt.session.max_awaiting_rel = 100

## Awaiting PUBREL Timeout
mqtt.session.await_rel_timeout = 20s

## Enable Statistics: on | off
mqtt.session.enable_stats = off

## Expired after 1 day:
## w - week
## d - day
## h - hour
## m - minute
## s - second
mqtt.session.expiry_interval = 2h

## Ignore message from self publish
mqtt.session.ignore_loop_deliver = false

##--------------------------------------------------------------------
## MQTT Message Queue
##--------------------------------------------------------------------

## Type: simple | priority
mqtt.mqueue.type = simple

## Topic Priority: 0~255, Default is 0
## mqtt.mqueue.priority = topic/1=10,topic/2=8

## Max queue length. Enqueued messages when persistent client disconnected,
## or inflight window is full. 0 means no limit.
mqtt.mqueue.max_length = 1000

## Low-water mark of queued messages
mqtt.mqueue.low_watermark = 20%

## High-water mark of queued messages
mqtt.mqueue.high_watermark = 60%

## Queue Qos0 messages?
mqtt.mqueue.store_qos0 = true

##--------------------------------------------------------------------
## MQTT Broker and PubSub
##--------------------------------------------------------------------

## System Interval of publishing broker $SYS Messages
mqtt.broker.sys_interval = 1m

## PubSub Pool Size. Default should be scheduler numbers.
mqtt.pubsub.pool_size = 8

# mqtt.pubsub.by_clientid = true

## Subscribe Asynchronously
mqtt.pubsub.async = true

##--------------------------------------------------------------------
## MQTT Bridge
##--------------------------------------------------------------------

## Bridge Queue Size
mqtt.bridge.max_queue_len = 10000

## Ping Interval of bridge node. Unit: Second
mqtt.bridge.ping_down_interval = 1s

##-------------------------------------------------------------------
## MQTT Plugins
##-------------------------------------------------------------------

## Dir of plugins' config
mqtt.plugins.etc_dir =/etc/emqttd/plugins/

## File to store loaded plugin names.
mqtt.plugins.loaded_file = /var/lib/emqttd/loaded_plugins

##--------------------------------------------------------------------
## MQTT Listeners
##--------------------------------------------------------------------

##--------------------------------------------------------------------
## External TCP Listener

## External TCP Listener: 1883, 127.0.0.1:1883, ::1:1883
##listener.tcp.external = 11883
listener.tcp.external = 11883

## Size of acceptor pool
listener.tcp.external.acceptors = 16

## Maximum number of concurrent clients
listener.tcp.external.max_clients = 102400

#listener.tcp.external.mountpoint = external/

## Rate Limit. Format is 'burst,rate', Unit is KB/Sec
#listener.tcp.external.rate_limit = 100,10

#listener.tcp.external.access.1 = allow 192.168.0.0/24

listener.tcp.external.access.2 = allow all

## Proxy Protocol V1/2
## listener.tcp.external.proxy_protocol = on
## listener.tcp.external.proxy_protocol_timeout = 3s

## TCP Socket Options
listener.tcp.external.backlog = 1024

#listener.tcp.external.recbuf = 4KB

#listener.tcp.external.sndbuf = 4KB

listener.tcp.external.buffer = 4KB

listener.tcp.external.nodelay = true

##--------------------------------------------------------------------
## Internal TCP Listener

## Internal TCP Listener: 11883, 127.0.0.1:11883, ::1:11883
listener.tcp.internal = 127.0.0.1:1883

## Size of acceptor pool
listener.tcp.internal.acceptors = 16

## Maximum number of concurrent clients
listener.tcp.internal.max_clients = 102400

#listener.tcp.external.mountpoint = internal/

## Rate Limit. Format is 'burst,rate', Unit is KB/Sec
## listener.tcp.internal.rate_limit = 1000,100

## TCP Socket Options
listener.tcp.internal.backlog = 512

listener.tcp.internal.tune_buffer = on

listener.tcp.internal.buffer = 1MB

listener.tcp.internal.recbuf = 4KB

listener.tcp.internal.sndbuf = 1MB

listener.tcp.internal.nodelay = true

##--------------------------------------------------------------------
## External SSL Listener

## SSL Listener: 8883, 127.0.0.1:8883, ::1:8883
listener.ssl.external = 18883

## Size of acceptor pool
listener.ssl.external.acceptors = 16

## Maximum number of concurrent clients
listener.ssl.external.max_clients = 1024

## listener.ssl.external.mountpoint = inbound/

## Rate Limit. Format is 'burst,rate', Unit is KB/Sec
## listener.ssl.external.rate_limit = 100,10

## Proxy Protocol V1/2
## listener.ssl.external.proxy_protocol = on
## listener.ssl.external.proxy_protocol_timeout = 3s

listener.ssl.external.access.1 = allow all

### SSL Options. See http://erlang.org/doc/man/ssl.html

## Configuring SSL Options. See http://erlang.org/doc/man/ssl.html
### TLS only for POODLE attack
## listener.ssl.external.tls_versions = tlsv1.2,tlsv1.1,tlsv1

### The Ephemeral Diffie-Helman key exchange is a very effective way of
### ensuring Forward Secrecy by exchanging a set of keys that never hit
### the wire. Since the DH key is effectively signed by the private key,
### it needs to be at least as strong as the private key. In addition, 
### the default DH groups that most of the OpenSSL installations have 
### are only a handful (since they are distributed with the OpenSSL
### package that has been built for the operating system it’s running on)
### and hence predictable (not to mention, 1024 bits only).

### In order to escape this situation, first we need to generate a fresh,
### strong DH group, store it in a file and then use the option above,
### to force our SSL application to use the new DH group. Fortunately,
### OpenSSL provides us with a tool to do that. Simply run:
### openssl dhparam -out dh-params.pem 2048

listener.ssl.external.handshake_timeout = 15s

listener.ssl.external.keyfile = /etc/ssl/private/myserver.mydomain.nl.key.pem

listener.ssl.external.certfile = /etc/ssl/certs/myserver.mydomain.nl.chain.pem

## listener.ssl.external.cacertfile = /etc/emqttd/certs/cacert.pem

## listener.ssl.external.dhfile = /etc/emqttd/certs/dh-params.pem

## listener.ssl.external.verify = verify_peer

## listener.ssl.external.fail_if_no_peer_cert = true

### This is the single most important configuration option of an Erlang SSL application.
### Ciphers (and their ordering) define the way the client and server encrypt information
### over the wire, from the initial Diffie-Helman key exchange, the session key encryption
### algorithm and the message digest algorithm. Selecting a good cipher suite is critical
### for the application’s data security, confidentiality and performance.
### The cipher list above offers:
### 
### A good balance between compatibility with older browsers. It can get stricter for Machine-To-Machine scenarios.
### Perfect Forward Secrecy.
### No old/insecure encryption and HMAC algorithms
###
### Most of it was copied from Mozilla’s Server Side TLS article
## listener.ssl.external.ciphers = ECDHE-ECDSA-AES256-GCM-SHA384,ECDHE-RSA-AES256-GCM-SHA384,ECDHE-ECDSA-AES256-SHA384,ECDHE-RSA-AES256-SHA384,ECDHE-ECDSA-DES-CBC3-SHA,ECDH-ECDSA-AES256-GCM-SHA384,ECDH-RSA-AES256-GCM-SHA384,ECDH-ECDSA-AES256-SHA384,ECDH-RSA-AES256-SHA384,DHE-DSS-AES256-GCM-SHA384,DHE-DSS-AES256-SHA256,AES256-GCM-SHA384,AES256-SHA256,ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-GCM-SHA256,ECDHE-ECDSA-AES128-SHA256,ECDHE-RSA-AES128-SHA256,ECDH-ECDSA-AES128-GCM-SHA256,ECDH-RSA-AES128-GCM-SHA256,ECDH-ECDSA-AES128-SHA256,ECDH-RSA-AES128-SHA256,DHE-DSS-AES128-GCM-SHA256,DHE-DSS-AES128-SHA256,AES128-GCM-SHA256,AES128-SHA256,ECDHE-ECDSA-AES256-SHA,ECDHE-RSA-AES256-SHA,DHE-DSS-AES256-SHA,ECDH-ECDSA-AES256-SHA,ECDH-RSA-AES256-SHA,AES256-SHA,ECDHE-ECDSA-AES128-SHA,ECDHE-RSA-AES128-SHA,DHE-DSS-AES128-SHA,ECDH-ECDSA-AES128-SHA,ECDH-RSA-AES128-SHA,AES128-SHA

### SSL parameter renegotiation is a feature that allows a client and 
### a server to renegotiate the parameters of the SSL connection on the fly.
### RFC 5746 defines a more secure way of doing this. By enabling secure renegotiation,
### you drop support for the insecure renegotiation, prone to MitM attacks. 
## listener.ssl.external.secure_renegotiate = off

### A performance optimization setting, it allows clients to reuse 
### pre-existing sessions, instead of initializing new ones.
### Read more about it here.
## listener.ssl.external.reuse_sessions = on

### An important security setting, it forces the cipher to be set based on
### the server-specified order instead of the client-specified order,
### hence enforcing the (usually more properly configured) security
### ordering of the server administrator.
## listener.ssl.external.honor_cipher_order = on

### Use the CN or DN value from the client certificate as a username.
### Notice: 'verify' should be configured as 'verify_peer'
## listener.ssl.external.peer_cert_as_username = cn

## SSL Socket Options
## listener.ssl.external.backlog = 1024

## listener.ssl.external.recbuf = 4KB

## listener.ssl.external.sndbuf = 4KB

## listener.ssl.external.buffer = 4KB

## listener.ssl.external.nodelay = true

##--------------------------------------------------------------------
## External MQTT/WebSocket Listener

listener.ws.external = 18083

listener.ws.external.acceptors = 4

listener.ws.external.max_clients = 64

listener.ws.external.access.1 = allow all

## TCP Options
listener.ws.external.backlog = 1024

listener.ws.external.recbuf = 4KB

listener.ws.external.sndbuf = 4KB

listener.ws.external.buffer = 4KB

listener.ws.external.nodelay = true

##--------------------------------------------------------------------
## External MQTT/WebSocket/SSL Listener

listener.wss.external = 18084

listener.wss.external.acceptors = 4

listener.wss.external.max_clients = 64

listener.wss.external.access.1 = allow all

## SSL Options
listener.wss.external.handshake_timeout = 15s

listener.wss.external.keyfile = /etc/ssl/private/myserver.mydomain.nl.key.pem

listener.wss.external.certfile = /etc/ssl/certs/myserver.mydomain.nl.chain.pem

## listener.wss.external.cacertfile = /etc/emqttd/certs/cacert.pem

## listener.wss.external.verify = verify_peer

## listener.wss.external.fail_if_no_peer_cert = true

##--------------------------------------------------------------------
## HTTP Management API Listener

#listener.api.mgmt = 127.0.0.1:8088
listener.api.mgmt = 0.0.0.0:8088

listener.api.mgmt.acceptors = 4

listener.api.mgmt.max_clients = 64

listener.api.mgmt.access.1 = allow all

##-------------------------------------------------------------------
## System Monitor
##-------------------------------------------------------------------

## Long GC, don't monitor in production mode for:
## https://github.com/erlang/otp/blob/feb45017da36be78d4c5784d758ede619fa7bfd3/erts/emulator/beam/erl_gc.c#L421
sysmon.long_gc = false

## Long Schedule(ms)
sysmon.long_schedule = 240

## 8M words. 32MB on 32-bit VM, 64MB on 64-bit VM.
sysmon.large_heap = 8MB

## Busy Port
sysmon.busy_port = false

## Busy Dist Port
sysmon.busy_dist_port = true
```

**Comment by Gilbert-Wong** [comment]
It's weird. I tried to reproduce the problem you encountered. 
![image](https://user-images.githubusercontent.com/20479699/51477828-94473180-1dc4-11e9-80bc-48413490f7db.png) Look! It works well. According to the nslookup result your provide above, I see that the `myserver.mydomain.nl` has multiple corresponding ip addresses, `192.168.14.1` and `192.168.14.214`, which one is the actual ip address of the machine you run emqx on ?

**Comment by victorclaessen** [comment]
I also don't understand it (hence the question ;-)). 

When I replace the /etc/hosts file with the following, everything works as normal:
```
root@myserver:/# cat /etc/hosts
127.0.0.1       localhost
127.0.1.1      myserver.mydomain.nl myserver
#127.0.1.1       myserver
#192.168.14.224  myserver.mydomain.nl
```

Additionally, the ip ```192.168.14.1``` that you mention (in the output of nslookup) is the ip address of the DNS server. This is normal output for the nslookup program. 192.168.14.1 is not an ip of myserver.

**Comment by Gilbert-Wong** [comment]
`192.168.14.224` is that your server ip address? Replace your node name with `emq@192.168.14.224` and try again.

**Comment by victorclaessen** [comment]
Ah, that is a very good test!

I replaced the name in /etc/emqttd/emq.conf, and then restarted the node, and then this is the result:
```
root@myserver:/# emqttd_ctl status
Node 'emq@192.168.14.214' is started
emqttd 2.3.10 is running
```
This is now independent of what is in /etc/hosts (i.e. both situations work, 127.0.1.1 myserver.mydomain.nl or 192.168.14.224 myserver.mydomain.nl , it doesn't matter)

```
root@myserver:/# emqttd_ctl cluster status
Cluster status: [{running_nodes,['emq@otherserver.mydomain.nl',
                                 'emq@192.168.14.214']},
                 {stopped_nodes,['emq@myserver.mydomain.nl']}]
```

The question is: why does it work with the ip address 192.168.14.214, but not with the hostname (myserver.mydomain.nl) that resolves to 192.168.14.214 (it resolves both with /etc/hosts, and also with DNS)? That seems so weird to me.

**Comment by Gilbert-Wong** [comment]
It's weird to me too because I can not reproduce your problem on my machine, I have a question, did you start your emqx first then config dns resolve in /etc/hosts ?

**Comment by victorclaessen** [comment]
```
(emq@192.168.14.214)1> inet:gethostbyname("myserver.mydomain.nl").
{ok,{hostent,"myserver",
             ["myserver.mydomain.nl"],
             inet,4,
             [{192,168,14,224}]}}
```

**Comment by victorclaessen** [comment]
> It's weird to me too because I can not reproduce your problem on my machine, I have a question, did you start your emqx first, then config dns resolve in /etc/hosts ?

I restarted emqx/emqttd after changing /etc/hosts.

**Comment by victorclaessen** [comment]
Also interesting: when I change the node name back to myserver.mydomain.nl, and restart the emqttd console, then the ```inet:gethostbyname("myserver.mydomain.nl")``` command doesn't produce **_any_** result.

**Comment by Gilbert-Wong** [comment]
Are you sure? @victorclaessen , this command must return the value `{ok, Hostent}` or `{error, posix()}`.

**Comment by Gilbert-Wong** [comment]
![image](https://user-images.githubusercontent.com/20479699/51503835-596ee900-1e18-11e9-940a-0b78e0ac465a.png)
Hey, look the result. The myserver.mydomain.nl was resolved to `192.168.14.224`, not `192.168.14.214`

**Comment by spring2maz** [comment]
Hi @victorclaessen I am a bit confused with your IP settings.

Which IP does myserver.mydomain.nl is intended to be resolved to exactly ?

If it is to be 192,168,14,214, please make sure `emq@192.168.14.214` is stopped before starting `emq@myserver.mydomain.nl` because Erlang does not allow same node name on the same host.

If 192.168.14.214 is where you have a running node, but NOT `myserver`, also 192,168,14,224 is indeed intended for `myserver`, please try the `emqttd_ctl` commands on `myserver` i.e. `224` instead of on `214` (remotely).
We would like to see if the command works locally, then try to troubleshoot why it does not work remotely.

**Comment by victorclaessen** [comment]
> Hey, look the result. The myserver.mydomain.nl was resolved to `192.168.14.224`, not `192.168.14.214`

Oh, man, you are so right. **214** is the correct ip address, and was I mistyping **224** into /etc/hosts. So emqttd_ctl could not resolve the hostname to the correct ip address.

I am so sorry for having wasted your time. Thank you very much for the support!
